src/subpages/list.tsx

```diff
@@ -21,48 +29,92 @@ const ROUND_DIGITS = 4;
-              data && data.chainQuery.blockQuery && data.chainQuery.blockQuery.blocks
-                ? (data.chainQuery.blockQuery.blocks as Block[])
-                : null;
-          }
-
-          return (
-            <>
-              <SummaryCards blocks={blocks} />
-              <OffsetSwitch
-                olderHandler={olderHandler}
-                newerHandler={newerHandler}
-                disable={{ older: loading, newer: loading || offset < 1 }}
-              />
-              <BlockList
-                blocks={blocks}
-                loading={loading}
-                columns={(mainMineColumns(props.pageContext.endpoint.name))}
-                endpointName={props.pageContext.endpoint.name}
-              />
-            </>
-          );
+      <SummaryCards blocks={blocks} />
+      <OffsetSwitch
+        olderHandler={olderHandler}
+        newerHandler={newerHandler}
+        disable={{
+          older: blocksLoading || transactionsLoading,
+          newer: blocksLoading || transactionsLoading || offset < 1,
         }}
-      </BlockListComponent>
+      />
+      <Pivot>
+        <PivotItem headerText='Blocks'>
+          <BlockListComponent
+            variables={{ offset, limit, excludeEmptyTxs }}
+            pollInterval={POLL_INTERVAL}>
+            {({ data, loading, error }) => {
+              setBlocksLoading(loading);
+              if (error) {
+                console.error(error);
+                return <p>{error.message}</p>;
+              }
+
+              if (!loading) {
+                setBlocks(
+                  data &&
+                    data.chainQuery.blockQuery &&
+                    data.chainQuery.blockQuery.blocks
+                    ? (data.chainQuery.blockQuery.blocks as Block[])
+                    : null
+                );
+              }
+
+              return (
+                <BlockList
+                  blocks={blocks}
+                  loading={loading}
+                  columns={(mainMineColumns(props.pageContext.endpoint.name))}
+                  endpointName={props.pageContext.endpoint.name}
+                />
+              );
+            }}
+          </BlockListComponent>
+        </PivotItem>
+        <PivotItem headerText='Transactions'>
+          <TransactionListComponent
+            variables={{ offset, limit, desc: true }}
+            pollInterval={POLL_INTERVAL}>
+            {({ data, loading, error }) => {
+              setTransactionsLoading(loading);
+              if (error) {
+                console.error(error);
+                return <p>{error.message}</p>;
+              }
+
+              let transactions = null;
+              if (!loading) {
+                transactions =
+                  data &&
+                  data.chainQuery.transactionQuery &&
+                  data.chainQuery.transactionQuery.transactions
+                    ? (data.chainQuery.transactionQuery
+                        .transactions as Transaction[])
```

Comment:
```suggestion
                        .transactions!)
```

This will also work if you are doing this to avoid null-related checks.

Reply:
Actually this seems to be here for type checks, so I think I'm going to have it like this for the moment:

```
Argument of type '({ __typename?: "Block" | undefined; } & Pick<Block, "miner"> & { previousBlock: Maybe<{ __typename?: "Block" | undefined; } & Pick<Block, "timestamp">>; transactions: ({ ...; } & Pick<...>)[]; } & { ...; } & Pick<...>)[] | null' is not assignable to parameter of type 'SetStateAction<Block[] | null>'.
  Type '({ __typename?: "Block" | undefined; } & Pick<Block, "miner"> & { previousBlock: Maybe<{ __typename?: "Block" | undefined; } & Pick<Block, "timestamp">>; transactions: ({ ...; } & Pick<...>)[]; } & { ...; } & Pick<...>)[]' is not assignable to type 'SetStateAction<Block[] | null>'.
    Type '({ __typename?: "Block" | undefined; } & Pick<Block, "miner"> & { previousBlock: Maybe<{ __typename?: "Block" | undefined; } & Pick<Block, "timestamp">>; transactions: ({ ...; } & Pick<...>)[]; } & { ...; } & Pick<...>)[]' is not assignable to type 'Block[]'.
      Property 'nonce' is missing in type '{ __typename?: "Block" | undefined; } & Pick<Block, "miner"> & { previousBlock: Maybe<{ __typename?: "Block" | undefined; } & Pick<Block, "timestamp">>; transactions: ({ ...; } & Pick<...>)[]; } & { ...; } & Pick<...>' but required in type 'Block'.ts(2345)
graphql.tsx(46, 3): 'nonce' is declared here.```